### PR TITLE
Require strong password in forgot password form

### PR DESCRIPTION
**Pull request by panoschal**
Fixes https://github.com/vector-im/element-web/issues/9557

Use the PassphraseField component in ForgotPassword form, instead of a plain Field, in order to check password complexity requirements.

Display an error when the new password is not strong enough.

The validation logic is not the same as in registration, it displays an error dialog when you submit, instead of focusing the invalid field.
I thought it would be better to keep this PR short, so I kept it as it was.

I can make it focus the invalid fields in another one, if it is desirable.

Signed-off-by: Panagiotis Chalimourdas <panagiotis4531@gmail.com>

### CODE REVIEW

**Comment by SimonBrandner** [comment]
@panoschal, thanks for working on this! Could you please include a sign-off as described [here](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.rst#sign-off)?

**Comment by panoschal** [comment]
> @panoschal, thanks for working on this! Could you please include a sign-off as described [here](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.rst#sign-off)?

Yes, sorry, I appended to the description.

File: src/components/structures/auth/ForgotPassword.js [hunk 1]
```diff
@@ -27,8 +27,11 @@ import classNames from 'classnames';
 import AuthPage from "../../views/auth/AuthPage";
 import CountlyAnalytics from "../../../CountlyAnalytics";
 import ServerPicker from "../../views/elements/ServerPicker";
+import PassphraseField from '../../views/auth/PassphraseField';
 import {replaceableComponent} from "../../../utils/replaceableComponent";
 
+const PASSWORD_MIN_SCORE = 3; // safely unguessable: moderate protection from offline slow-hash scenario.
```
Comment (turt2live) [inline src/components/structures/auth/ForgotPassword.js:33]: we should share this with the registration stuff (export it from there and import here, if it doesn't explode)

**Review by turt2live — APPROVED** [review]
Otherwise this looks great, thank you!

File: src/components/structures/auth/ForgotPassword.js (hunk 1 above)
Comment (panoschal) [reply src/components/structures/auth/ForgotPassword.js]: great!

**Review by turt2live — APPROVED** [review]
Thanks!

**Pull request merged**